hummingbot/connector/exchange/bloxroute_openbook/bloxroute_openbook_exchange.py

```diff
@@ -0,0 +1,533 @@
+        self._sol_wallet_public_key = solana_wallet_public_key
+        self._sol_wallet_private_key = solana_wallet_private_key
+        self._trading_required = trading_required
+        self._hummingbot_to_solana_id = {}
+        self._open_orders_address = open_orders_address
+
+        self._server_response = GetServerTimeResponse
+        endpoint = CONSTANTS.WS_URL
+        self._provider_1: Provider = WsProvider(endpoint=endpoint, auth_header=self._auth_header, private_key=self._sol_wallet_private_key)
+        self._provider_2: Provider = WsProvider(endpoint=endpoint, auth_header=self._auth_header, private_key=self._sol_wallet_private_key)
+
+        self._trading_pairs = trading_pairs
+        self._order_manager: BloxrouteOpenbookOrderDataManager = BloxrouteOpenbookOrderDataManager(
+            self._provider_2, self._trading_pairs, self._sol_wallet_public_key
+        )
+        self._order_book_manager_connected = False
+        asyncio.create_task(self._initialize_order_manager())
+
+        super().__init__(client_config_map)
+        self.real_time_balance_update = False
+
+    async def _initialize_order_manager(self):
+        await self._order_manager.start()
+        self._order_book_manager_connected = True
+
+        print("order books initialized!")
+
+    def authenticator(self):
+        return BloxrouteOpenbookAuth(
+            api_key=self._auth_header, secret_key=self._sol_wallet_private_key, time_provider=self._time_synchronizer
+        )
+
+    @property
+    def name(self) -> str:
+        return CONSTANTS.EXCHANGE_NAME
+
+    async def check_network(self) -> NetworkStatus:
+        await self._provider_1.connect()
+        await self._order_manager.start()
+
+        try:
+            self._server_response: GetServerTimeResponse = await self._provider_1.get_server_time()
+            if self._server_response.timestamp:
+                return NetworkStatus.CONNECTED
+            else:
+                return NetworkStatus.NOT_CONNECTED
+        except Exception:
+            return NetworkStatus.NOT_CONNECTED
+
+    async def _update_time_synchronizer(self, pass_on_non_cancelled_error: bool = False):
+        pass
+
+    @property
+    def status_dict(self) -> Dict[str, bool]:
+        return {
+            "order_books_initialized": self._order_book_manager_connected,
+            "trading_rule_initialized": len(self._trading_rules) != 0,
+        }
+
+    def get_price(self, trading_pair: str, is_buy: bool) -> Decimal:
+        if self._order_manager.is_ready:
+            price, _ = self._order_manager.get_price_with_opportunity_size(trading_pair=trading_pair, is_buy=is_buy)
+            return Decimal(price)
+        else:
+            if not self._order_manager.started:
+                asyncio.create_task(self._initialize_order_manager())
+            return Decimal(0)
+
+    @property
+    def rate_limits_rules(self):
+        return CONSTANTS.RATE_LIMITS
+
+    @property
+    def domain(self):
```

Comment:
since you're not really making use of most of these anyway I'd suggest deleting most of these properties. just too much clutter

Reply:
I think these are called by ExchangePyBase directly in `__init__` or `buy` or `sell` and they would throw exceptions if they weren't implemented